core/dbt/adapters/base/connections.py

```diff
@@ -142,44 +142,44 @@ def exception_handler(self, sql: str) -> ContextManager:
         )
 
     def set_connection_name(self, name: Optional[str] = None) -> Connection:
-        conn_name: str
-        if name is None:
-            # if a name isn't specified, we'll re-use a single handle
-            # named 'master'
-            conn_name = "master"
-        else:
-            if not isinstance(name, str):
-                raise dbt.exceptions.CompilerException(
-                    f"For connection name, got {name} - not a string!"
-                )
-            assert isinstance(name, str)
-            conn_name = name
+        """Called by 'acquire_connection' in BaseAdapter, which is called by
+        'connection_named', called by 'connection_for(node)'.
+        Creates a connection for this thread if one doesn't already
+        exist, and will rename an existing connection."""
 
+        conn_name: str = "master" if name is None else name
```

Comment:
Type hint should be `: str | None` or `: Optional[str]`

Reply:
It can't be Optional[str] because the logging events are defined in protobufs which don't have Optional[str].

Reply:
Ah I misread it!